src/App.tsx

```diff
@@ -1,18 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [curentPage, setCurentPage] = useState(1);
+
+  const total = items.length;
+  const visibleItemStartIndex = perPage * curentPage - perPage;
+  const visibleItemEndIndex = perPage * curentPage;
+  const visibleItems = items.slice(visibleItemStartIndex, visibleItemEndIndex);
+  const paginationInfo = `Page ${curentPage} (${perPage * curentPage
+    - perPage + 1} - ${perPage * curentPage < total ? perPage * curentPage : total} of ${total})`;
```

Comment:
this one is a no-go. Totally not clear what's happening here and what the output might be. We can't have it this way in our code